driver-scala/src/main/scala/org/mongodb/scala/ObservableImplicits.scala

```diff
@@ -116,17 +118,41 @@ trait ObservableImplicits {
     override def subscribe(observer: Observer[_ >: GridFSFile]): Unit = Mono.from(publisher).subscribe(observer)
   }
 
-  implicit class ToSingleObservableVoid(pub: => Publisher[Void]) extends SingleObservable[Void] {
```

Comment:
`ToSingleObservableVoid` is part of public API (https://mongodb.github.io/mongo-java-driver/4.11/apidocs/mongo-scala-driver/org/mongodb/scala/ObservableImplicits$ToSingleObservableVoid.html), but my understanding is that

1. There was never a reason for any user to use it explicitly (though I found a single explicit import here https://github.com/hmrc/event-hub/blob/main/app/uk/gov/hmrc/eventhub/service/TransactionHandlerImpl.scala#L19).
2. The implicit conversion was convenient for the driver code, and is completely unneeded for any user of the Scala driver API, because it does not expose `Publisher[Void]`. I wonder if `ToSingleObservableVoid` is part of public API simply because there is no way to remove it from there, even though users don't need it.

For the aforementioned reasons, it seems to me that removing `ToSingleObservableVoid` and introducing `ToSingleObservableUnit` is fine (they can't coexist).

Reply:
Yeap its an implicit class, so using it directly is strange. However, having it in scope makes it to be usable in customers code.  Either way this is a breaking change but fixes a bug so makes sense.